scripts/npm/create-component.js

```diff
@@ -0,0 +1,206 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+/* eslint-disable no-console */
+
+const { execSync } = require('child_process');
+const prompt = require('prompt');
+const fs = require('fs');
+const colors = require('colors');
+const { Transform } = require('stream');
+const path = require('path');
+const _ = require('lodash');
+const globby = require('globby');
+
+const STORYBOOK_CONFIG_SPLIT_POINT_1 = 'configure(() => {';
+const STORYBOOK_CONFIG_SPLIT_POINT_2 = '}, module);';
+
+const schema = {
+  properties: {
+    platform: {
+      description: 'native or web?',
+      default: 'native',
+      pattern: /^(native|web)$/,
+      message: 'Enter "native" or "web"',
+    },
+    name: {
+      description: "Enter the component name, e.g. 'banner-alert'",
+      pattern: /^[a-z-]+$/,
+      message: 'Use snake case, e.g. "banner-alert".',
+      required: true,
+    },
+  },
+};
+
+_.mixin({
+  pascalCase: _.flow(
+    _.camelCase,
+    _.upperFirst,
+  ),
+});
+
+// Util to recursively make dirs
+const mkdirp = dir =>
+  path
+    .resolve(dir)
+    .split(path.sep)
+    .reduce((acc, cur) => {
+      const currentPath = path.normalize(acc + path.sep + cur);
+      try {
+        fs.statSync(currentPath);
+      } catch (e) {
+        if (e.code === 'ENOENT') {
+          fs.mkdirSync(currentPath);
+        } else {
+          throw e;
+        }
+      }
+      return currentPath;
+    }, '');
+
+const Replacer = (source, destination) =>
+  new Transform({
+    transform(chunk, enc, cb) {
+      const data = chunk.toString();
+      this.push(data.replace(new RegExp(source, 'g'), destination));
+      cb();
+    },
+  });
+
+const sortLines = (lineA, lineB) => {
+  const trimmedA = lineA.trim();
+  const trimmedB = lineB.trim();
+  if (trimmedA < trimmedB) {
+    return -1;
+  }
+  if (trimmedA > trimmedB) {
+    return 1;
+  }
+  return 0;
+};
+
+const createComponent = async (err, { platform, name }) => {
+  if (err) {
+    console.error(err);
+    return;
+  }
+
+  let boilerplateComponentPath = `packages/bpk-component-boilerplate`;
```

Comment:
`let` 😢 

Would it be nicer to write it like this? I appreciate not everyone is as alergic to `let` as I am
```javascript
const { boilerplateComponentPath, newComponentPath, storybookConfigFile, storybookImport } = platform === 'native' ? {
  boilerplateComponentPath:`native/packages/react-native-bpk-component-boilerplate`,
  newComponentPath: `native/packages/react-native-bpk-component-${name}`,
  storybookConfigFile: `native/storybook/storybook.js`,
  storybookImport:`require('../packages/react-native-bpk-component-${name}/stories');`,
} : {
  boilerplateComponentPath:`packages/bpk-component-boilerplate`,
  newComponentPath: `packages/bpk-component-${name}`,
  storybookConfigFile: `./storybook/config.js`,
  storybookImport:`require('./../${newComponentPath}/stories');`,
}
```

```

Reply:
TBH I don't like that at all, there are a few use cases for let and this one is probably a good one, way more readable than the ternary operator